osgp/platform/osgp-core/src/main/java/org/opensmartgridplatform/core/infra/jms/protocol/outbound/ProtocolRequestMessageJmsTemplateFactory.java

```diff
@@ -45,7 +45,18 @@ public ProtocolRequestMessageJmsTemplateFactory(
   }
 
   public JmsTemplate getJmsTemplate(final ProtocolInfo protocolInfo) {
-    return this.jmsTemplateRegistry.getValue(protocolInfo.getKey());
+    if (this.jmsTemplateRegistry.getValue(protocolInfo.getKey()) != null) {
+      return this.jmsTemplateRegistry.getValue(protocolInfo.getKey());
+    }
+    return this.jmsTemplateRegistry.getValue(this.keyWithoutVariant(protocolInfo));
+  }
+
+  private String keyWithoutVariant(final ProtocolInfo protocolInfo) {
+    return new ProtocolInfo.Builder()
+        .withProtocol(protocolInfo.getProtocol())
+        .withProtocolVersion(protocolInfo.getProtocolVersion())
+        .build()
+        .getKey();
   }
 
```

Comment:
Ik zou hier eigenlijk nog een test bij verwachten. Ik zie dat er zo wie zo weinig testen zijn op dit stuk. Ik vraag me dus ook af hoe we hiermee om gaan.

Reply:
De test is toegevoegd door Gerben.